Update windows install with Winget and new exe (#4506)
grafana/alloy: docs/sources/set-up/install/windows.md

# Install {{% param "FULL_PRODUCT_NAME" %}} on Windows

You can install {{< param "PRODUCT_NAME" >}} on Windows as a standard graphical install, a WinGet install, or a silent install.

## Standard graphical install

To do a standard graphical install of {{< param "PRODUCT_NAME" >}} on Windows, perform the following steps.

1. Navigate to the [releases page][releases] on GitHub.
1. Scroll down to the **Assets** section.

> 1. Download and extract `alloy-installer-windows-amd64.exe.zip` or download `alloy-installer-windows-amd64.exe`.

Reviewer: I would swap the order here, as the former is more steps for the user to do.

1. Double-click on `alloy-installer-windows-amd64.exe` to install {{< param "PRODUCT_NAME" >}}.

The installer places {{< param "PRODUCT_NAME" >}} in the default directory `%PROGRAMFILES%\GrafanaLabs\Alloy`.

## WinGet install

To install {{< param "PRODUCT_NAME" >}} with WinGet, perform the following steps.

> 1. Make sure that the [WinGet package manager](https://learn.microsoft.com/en-gb/windows/package-manager/winget/) is available.

Reviewer: ```suggestion
1. Make sure that the [WinGet package manager](https://learn.microsoft.com/windows/package-manager/winget/) is installed.
```

> 1. Run the following command as Administrator in PowerShell.

Reviewer: It is not necessary to run as an administrator, as the installer will elevate to request permission if required.

This can also be done in any terminal for Windows, it doesn't have to be from PowerShell.

   ```powershell
   winget install GrafanaLabs.Alloy
   ```

## Silent install

To do a silent install of {{< param "PRODUCT_NAME" >}} on Windows, perform the following steps.

1. Navigate to the [releases page][releases] on GitHub.
1. Scroll down to the **Assets** section.

> 1. Download and extract `alloy-installer-windows-amd64.exe.zip` or download `alloy-installer-windows-amd64.exe`.

Reviewer: Same here regarding swapping the order.

1. Double-click on `alloy-installer-windows-amd64.exe` to install {{< param "PRODUCT_NAME" >}}.

The installer places {{< param "PRODUCT_NAME" >}} in the default directory `%PROGRAMFILES%\GrafanaLabs\Alloy`.

## WinGet install

To install {{< param "PRODUCT_NAME" >}} with WinGet, perform the following steps.

1. Make sure that the [WinGet package manager](https://learn.microsoft.com/windows/package-manager/winget/) is installed.
1. Run the following command.

   {{< code >}}

   ```cmd
   winget install GrafanaLabs.Alloy
   ```

   ```powershell
   winget install GrafanaLabs.Alloy
   ```

   {{< /code >}}

## Silent install

To do a silent install of {{< param "PRODUCT_NAME" >}} on Windows, perform the following steps.

1. Navigate to the [releases page][releases] on GitHub.
1. Scroll down to the **Assets** section.
1. Download `alloy-installer-windows-amd64.exe` or download and extract `alloy-installer-windows-amd64.exe.zip`..
1. Run the following command as Administrator.

   {{< code >}}

   ```cmd
   <PATH>\alloy-installer-windows-amd64.exe /S
   ```

   ```powershell

>    <PATH>\alloy-installer-windows-amd64.exe /S

Reviewer: ```suggestion
   & <PATH>\alloy-installer-windows-amd64.exe /S
```

The `/RUNTIMEPRIORITY` installation option sets this flag, and if {{< param "PRODUCT_NAME" >}} isn't running with an appropriate stability level, it fails to start.

[stability]: https://grafana.com/docs/release-life-cycle/
[backward compatibility]: ../../../introduction/backward-compatibility/

{{< /admonition >}}

## Service configuration

{{< param "PRODUCT_NAME" >}} uses the Windows registry key `HKLM\Software\GrafanaLabs\Alloy` for service configuration.

- `Arguments`: Type `REG_MULTI_SZ`. Each value represents a binary argument for the {{< param "PRODUCT_NAME" >}} binary.
- `Environment`: Type `REG_MULTI_SZ`. Each value represents an environment value `KEY=VALUE` for the {{< param "PRODUCT_NAME" >}} binary.

## Uninstall

Uninstalling {{< param "PRODUCT_NAME" >}} stops the service and removes it from disk.
This includes any configuration files in the installation directory.

### Standard graphical uninstall

To uninstall {{< param "PRODUCT_NAME" >}}, use Add or Remove Programs or run the following command as Administrator.

{{< code >}}

```cmd
%PROGRAMFILES%\GrafanaLabs\Alloy\uninstall.exe
```

```powershell

> ${env:PROGRAMFILES}\GrafanaLabs\Alloy\uninstall.exe

Reviewer: ```suggestion
& ${env:PROGRAMFILES}\GrafanaLabs\Alloy\uninstall.exe
```

> ${env:PROGRAMFILES}\GrafanaLabs\Alloy\uninstall.exe

Reviewer: ```suggestion
& ${env:PROGRAMFILES}\GrafanaLabs\Alloy\uninstall.exe
```

```

{{< /code >}}

### Uninstall with WinGet

To install {{< param "PRODUCT_NAME" >}} with WinGet, run the following command.

{{< code >}}

```cmd
winget uninstall GrafanaLabs.Alloy
```

```powershell
winget uninstall GrafanaLabs.Alloy
```

{{< /code >}}

### Silent uninstall

To silently uninstall {{< param "PRODUCT_NAME" >}}, run the following command as Administrator.

{{< code >}}

```cmd
%PROGRAMFILES%\GrafanaLabs\Alloy\uninstall.exe /S
```

```powershell

> ${env:PROGRAMFILES}\GrafanaLabs\Alloy\uninstall.exe /S

Reviewer: ```suggestion
& ${env:PROGRAMFILES}\GrafanaLabs\Alloy\uninstall.exe /S
```

The `/RUNTIMEPRIORITY` installation option sets this flag, and if {{< param "PRODUCT_NAME" >}} isn't running with an appropriate stability level, it fails to start.

[stability]: https://grafana.com/docs/release-life-cycle/
[backward compatibility]: ../../../introduction/backward-compatibility/

{{< /admonition >}}

## Service configuration

{{< param "PRODUCT_NAME" >}} uses the Windows registry key `HKLM\Software\GrafanaLabs\Alloy` for service configuration.

- `Arguments`: Type `REG_MULTI_SZ`. Each value represents a binary argument for the {{< param "PRODUCT_NAME" >}} binary.
- `Environment`: Type `REG_MULTI_SZ`. Each value represents an environment value `KEY=VALUE` for the {{< param "PRODUCT_NAME" >}} binary.

## Uninstall

Uninstalling {{< param "PRODUCT_NAME" >}} stops the service and removes it from disk.
This includes any configuration files in the installation directory.

### Standard graphical uninstall

To uninstall {{< param "PRODUCT_NAME" >}}, use Add or Remove Programs or run the following command as Administrator in PowerShell.

```powershell
%PROGRAMFILES%\GrafanaLabs\Alloy\uninstall.exe

> ```

Reviewer: This doesn't have to be done in PowerShell, and this isn't valid syntax for PowerShell.

For PowerShell the command would be:

```pwsh
& ${env:PROGRAMFILES}\GrafanaLabs\Alloy\uninstall.exe
```

### Uninstall with `winget`

> To install {{< param "PRODUCT_NAME" >}} with `winget`, run the following command as Administrator in PowerShell.

Reviewer: Same comment here regarding administrator and PowerShell.

```powershell
winget uninstall GrafanaLabs.Alloy
```

### Silent uninstall

To silently uninstall {{< param "PRODUCT_NAME" >}}, run the following command as Administrator in PowerShell.

```powershell

> %PROGRAMFILES%\GrafanaLabs\Alloy\uninstall.exe /S

Reviewer: Same here:

```pwsh
& ${env:PROGRAMFILES}\GrafanaLabs\Alloy\uninstall.exe /S
```
